source/common/upstream/cluster_manager_impl.cc

```diff
@@ -971,6 +971,11 @@ void ClusterManagerImpl::drainConnections() {
   });
 }
 
+bool ClusterManagerImpl::checkActiveStaticCluster(const std::string& cluster) {
+  const auto& it = active_clusters_.find(cluster);
```

Comment:
I'm on the fence on whether to make this function throw on error and retain the fidelity of error messages  or leave error handling in the call site and end up with a more generic error message.
I think I have a slight preference for more precise error messages. We can see if other reviewers have an opinion about it.

Reply:
agreed, I was was on the same on the fence, but I would slightly prefer leaving handling to the call site. Either way, the cause of the exception is clear for users - whether static or non active can be easily judged from their envoy.yaml.  @Shikugawa wdyt?

Reply:
I also agree with @yanavlasov 's idea.